plugins/modules/azure_rm_diagnosticsetting_info.py

```diff
@@ -0,0 +1,285 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2021 Alvin Ramoutar, (@AlvinRamoutar)
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_diagnosticsetting_info
+version_added: "0.1.0"
+short_description: Get Azure Diagnostic Setting info
+description:
+    - Get Azure Diagnostic Setting info.
+
+options:
+    resource_id:
+        description:
+            - The resource ID of the resource in which has the diagnostic setting.
+        required: true
+        type: str
```

Comment:
Argument 'resource_id' in argument_spec is not required, but is documented as being required

Reply:
Hey @Fred-sun ,
`resource_id` is indeed required, and is denoted as such in both documentation [and arg spec](https://github.com/AlvinRamoutar/azure/blob/b84724452b3ce66e74b4421be49f4cdf524ee8dd/plugins/modules/azure_rm_diagnosticsetting.py#L299),